symengine/utilities/matchpycpp/substitution.h

```diff
@@ -0,0 +1,58 @@
+#ifndef SYMENGINE_UTILITIES_MATCHPYCPP_SUBSTITUTION_H_
+#define SYMENGINE_UTILITIES_MATCHPYCPP_SUBSTITUTION_H_
+
+#include <symengine/basic.h>
+#include <symengine/add.h>
+#include <symengine/pow.h>
+#include <symengine/mul.h>
+
+#include <map>
+#include <vector>
+
+using namespace std;
+using namespace SymEngine;
+
+typedef map<string, multiset_basic> SubstitutionMultiset;
+
+int try_add_variable(SubstitutionMultiset &subst, const string variable_name,
```

Comment:
Use `bool` instead of `int`. Makes it easier to read and also convey what you are doing

Reply:
It was meant to resemble C's style exception handling. The Python implementation raises an exception.